### PR TITLE
fix: possible underflow in `_subtractTotalEarningSupply`

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: src/WrappedMToken.sol [hunk 1]
```diff
@@ -530,8 +530,15 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
...
+
+            if (amount_ >= totalEarningSupply_) _setTotalEarningSupply(0, 0);
+
+            uint112 principalOfTotalEarningSupply_ = _principalOfTotalEarningSupply;
             uint112 principal_ = IndexingMath.getPrincipalAmountRoundedDown(amount_, currentIndex_);
 
+            if (principal_ >= principalOfTotalEarningSupply_) _setTotalEarningSupply(0, 0);
+
             // Decrement the total earning supply and principal proportionally.
             _setTotalEarningSupply(totalEarningSupply() - amount_, _principalOfTotalEarningSupply - principal_);
```
Comment (PierrickGT) [inline src/WrappedMToken.sol:543]:
```suggestion
            _setTotalEarningSupply(totalEarningSupply() - amount_, principalOfTotalEarningSupply_ - principal_);
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

**Review by PierrickGT — APPROVED** [review]

File: src/WrappedMToken.sol [hunk 2]
```diff
@@ -530,10 +530,17 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
      */
     function _subtractTotalEarningSupply(uint240 amount_, uint128 currentIndex_) internal {
         unchecked {
+            uint240 totalEarningSupply_ = totalEarningSupply();
+
+            if (amount_ >= totalEarningSupply_) _setTotalEarningSupply(0, 0);
```
Comment (toninorair) [inline src/WrappedMToken.sol:535]: is the return missing here ?

**Review by toninorair — DISMISSED** [review]
let's briefly discuss this one before merge

File: src/WrappedMToken.sol (hunk 2 above)
Comment (deluca-mike) [reply src/WrappedMToken.sol:535]: Great catch! I missed the return on line 540 too!

**Pull request merged**